allauth/account/adapter.py

```diff
@@ -492,6 +493,7 @@ def authentication_failed(self, request, **credentials):
         dt = timezone.now()
         data.append(time.mktime(dt.timetuple()))
         cache.set(cache_key, data, app_settings.LOGIN_ATTEMPTS_TIMEOUT)
+        user_auth_failed.send(sender=self.__class__, request=request)
```

Comment:
A few thoughts:

- A signal is not needed per se, you can simply provide your own adapter and override the authentication_failed() method.

- If a signal is provided, it would be more consistent to name it after the adapter method (so authentication_failed, not user_auth_failed).

- To be truly useful, you would need to know details about the authentication attempt that failed -- as in, pass along the credentials into the signal.